presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -58,23 +60,26 @@ public double getOutputRowCount()
      * Returns estimated data size.
      * Unknown value is represented by {@link Double#NaN}
      */
-    public double getOutputSizeInBytes(Collection<Symbol> outputSymbols)
+    public double getOutputSizeInBytes(Collection<Symbol> outputSymbols, Map<Symbol, Type> types)
     {
         requireNonNull(outputSymbols, "outputSymbols is null");
 
         return outputSymbols.stream()
-                .map(this::getSymbolStatistics)
-                .mapToDouble(this::getOutputSizeForSymbol)
+                .mapToDouble(symbol -> getOutputSizeForSymbol(getSymbolStatistics(symbol), types.get(symbol)))
                 .sum();
     }
 
-    private double getOutputSizeForSymbol(SymbolStatsEstimate symbolStatistics)
+    private double getOutputSizeForSymbol(SymbolStatsEstimate symbolStatistics, Type type)
```

Comment:
validate `type` isn't null here (especially that is taken from a map, so can be null silently)